packages/web/pages/api/typedefs/article.ts

```diff
@@ -0,0 +1,174 @@
+                  const updatedArticle = updated.dataValues;
+                  const updatedData = {
+                    userId: currentUserId,
+                    articleId: updatedArticle.id,
+                    title: updatedArticle.title,
+                    body: updatedArticle.body,
+                    authorId: updatedArticle.authorId
+                  };
+
+                  if (
+                    articleModification?.dataValues &&
+                    moment(articleModification.dataValues.updatedAt).isSame(
+                      moment(),
+                      "minute"
+                    )
+                  ) {
+                    return articleModification
+                      .update(updatedData)
+                      .then(() => updated);
+                  } else {
+                    return db.article_modification
+                      .create(updatedData)
+                      .then(() => updated);
+                  }
+                })
+            )
+          )
+          .catch(err => {
+            throw new ApolloError(err);
+          })
+    ),
+    isFavourite: authenticated(
+      (_, { articleId }, { dataSources: { db }, currentUserId }) =>
+        db.favourite_article
+          .findOne({
+            where: { userId: currentUserId, articleId: articleId }
+          })
+          .then(favouriteArticle =>
+            favouriteArticle
+              ? favouriteArticle.destroy().then(() => false)
+              : db.favourite_article
+                  .create({
+                    userId: currentUserId,
+                    articleId: articleId
+                  })
+                  .then(() => true)
+          )
+    ),
+    moveArticle: authenticated(
+      (_, { id, parentId }, { dataSources: { db }, currentUserId }) =>
+        id === parentId
+          ? false
+          : db.article.findByPk(parentId).then(parent =>
+              getRootPath(parent, false).then(rootPath => {
+                if (rootPath.includes(id)) return false;
+                return db.article.findByPk(id).then(article => {
+                  if (article.dataValues.parentId == parentId) return false;
+                  return article
+                    .update({ parentId: parentId })
+                    .then(() => true);
+                });
+              })
+            )
+    )
+  },
+  Article: {
+    author: article => article.getAuthor(),
+    parent: article => article.getParent(),
+    children: article => article.getChildren(),
+    favourited: (article, args, { dataSources: { db }, currentUserId }) =>
+      db.favourite_article
+        .findOne({
+          where: { userId: currentUserId, articleId: article.id }
+        })
+        .then(favouriteArticle => !!favouriteArticle),
+    rootPath: article => getRootPath(article).then(path => path)
+  }
+};
+
+const moment = require("moment");
+
+const getRootPath = async (article, isSlug = true) => {
```

Comment:
This deserves it's own file.

Reply:
What do you mean with it's own file?, like a general file for functions?

I included it in the articles file since its strictly attached to the articles model